shared/src/main/scala/mlscript/ucs/Desugarer.scala

```diff
@@ -785,9 +786,39 @@ class Desugarer extends TypeDefs { self: Typer =>
       xs match {
         case MutCase.Constructor(className -> fields, cases) :: next =>
           printlnUCS(s"• Constructor pattern: $className(${fields.iterator.map(x => s"${x._1} -> ${x._2}").mkString(", ")})")
-          // TODO: expand bindings here
           val consequent = rec(cases)(defs ++ fields.iterator.map(_._2))
-          Case(className, mkLetFromFields(scrutinee, fields.toList, consequent), rec2(next))
+          val unapplyMtd = ctx.get(className.name) match {
+            case S(CompletedTypeInfo(cls: TypedNuCls)) => cls.td.genUnapply // Classes from other typing units
+            case S(ti: DelayedTypeInfo) => ti.decl.genUnapply // Classes in the same typing units
```

Comment:
I wonder why you check that the thing is a `Cls` in the line above but not in this one. Better be consistent!

Reply:
Removed class check in the first branch. It would be checked inside the `genUnapply`.